static/js/components/TruncatedText.js

```diff
@@ -31,9 +31,16 @@ export default function TruncatedText(props: Props) {
   const [expanded, setExpanded] = useState(false)
   const [hasOverflow, setHasOverflow] = useState(false)
   const [, forceReRender] = useState(0)
+  const [currentText, setCurrentText] = useState(text)
   const dotRef = useRef(null)
 
   useLayoutEffect(() => {
+    if (text !== currentText) {
+      setCurrentText(text)
+      setExpanded(false)
+      setHasOverflow(false)
+    }
```

Comment:
I think here if you wanted to you could just do this:

```js
useEffect(() => {
  setExpanded(false)
} [ text ])
```

which will undo the expansion whenever the `text` prop changes. then below in the `useLayoutEffect` we could just always set `hasOverflow` to the value of `container.scrollHeight > container.offsetHeight`

Reply:
I change the code to use useEffect() as you suggested.  The reset for hasOverflow is needed since dotdotdot is replaced by a plain div when the text is expanded

Reply:
right - I was just suggesting that in the existing `useLayoutEffect` callback we could change this line:

```js
if (container.scrollHeight > container.offsetHeight) {
    // scrollHeight is the height of the rendered text, while offsetHeight
    // is the height of the portion which is shown
    setHasOverflow(true)
}
```

to simply

```js
setHasOverflow(container.scrollHeight > container.offsetHeight)
```

Reply:
sorry my suggestion wasn't clear earlier :)

Reply:
although just resetting `hasOverflow(false)` whenever the `text` prop changes accomplishes basically the same thing - so if you want to stick with that I think that's fine

Reply:
That does not work because when the text is expanded `dotRef.current` is null (since there is no dotdotdot component` so there is no container

Reply:
Also I reverted the last commit. We don't want a separate useEffect() hook for the text because we want the code in useLayoutEffect() to run too when the text changes

Reply:
it does - when you call `useLayoutEffect` or `useEffect` and only pass the first argument (a function) it runs on every re-render

Reply:
if you want to merge this as-is it's fine with me - I just thought that if we could avoid the need for the currentText state we might want to, but it's nbd